docs/get-started/shader-modules.md

```diff
@@ -0,0 +1,184 @@
+# Shader Modules
+
+This tutorial will demonstrate how to use luma.gl shader modules to make reusable bits of functionality and dynamically insert them into your shaders. Most of this will be fairly similar to the [Hello Triangle](/docs/get-started/hello-triangle.md) app.
+
+We'll start by setting up our imports and defining our base vertex and fragment shaders:
+```js
+import {AnimationLoop, Model} from '@luma.gl/engine';
+import {Buffer, clear} from '@luma.gl/webgl';
+
+const vs1 = `
+  attribute vec2 position;
+
+  void main() {
+    gl_Position = vec4(position - vec2(0.5, 0.0), 0.0, 1.0);
+  }
+`;
+
+const fs1 = `
+  void main() {
+    gl_FragColor = color_getColor();
+  }
+`;
+
+const vs2 = `
+  attribute vec2 position;
+
+  void main() {
+    gl_Position = vec4(position + vec2(0.5, 0.0), 0.0, 1.0);
+  }
+`;
+
+const fs2 = `
+  void main() {
+    gl_FragColor = color_getColor() - 0.3;
+  }
+`;
+```
+
+We have two vertex and fragment shader pairs: one will move vertices to the left, the other moves vertices to the right. Both fragment shaders call a `color_getColor` function to set the final fragment color, but `color_getColor` isn't defined anywhere, so these shaders will not compile as-is.
+
+We define `color_getColor` in a shader module:
+```js
+const colorModule = {
+  name: "color",
+  fs: `
+    vec4 color_getColor() {
```

Comment:
Nit: should we show something more useful here, e.g. [HSL to RGB](https://github.com/uber/streetscape.gl/blob/master/modules/core/src/layers/point-cloud-layer/point-cloud-layer-vertex.glsl.js#L46)?